### PR TITLE
Adding IMPACT-Z.

### PR DESCRIPTION
**Pull request by hhslepicka**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hhslepicka** [comment]
Note for reviewer:
- In this recipe I didn't check the ` Source is from official source.` checkbox as I am currently working with the package maintainer upstream for them to start using releases of their code. So for now I tagged on my fork.
- I can't run a test command as this software has no `-help` or similar so on the tests I am just checking for the existence of the binary. This code simply runs the input in a folder so that is basically all I can do other than running the input which may take several minutes.

Thank you for your time and input.

**Comment by hhslepicka** [comment]
@ChristopherMayes could you please confirm that you accept to be a maintainer?

**Comment by ChristopherMayes** [comment]
I agree to be a maintainer

**Comment by hhslepicka** [comment]
@conda-forge/staged-recipes, this one is good for a review/merge.

**Comment by hhslepicka** [comment]
Pinging @conda-forge/staged-recipes for a review and merge. Thank you!

**Comment by hhslepicka** [comment]
@conda-forge-admin, please ping team

File: recipes/impact-z/build.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# for cross compiling using openmpi
+export OPAL_PREFIX=$PREFIX
+
+if [[ "$target_platform" == osx* ]]; then
+    CMAKE_FLAGS+=" -DCMAKE_OSX_SYSROOT=${CONDA_BUILD_SYSROOT}"
+    CMAKE_FLAGS+=" -DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET}"
```
Comment (isuruf) [inline recipes/impact-z/build.sh:8]:
```suggestion
```

**Review by isuruf — COMMENTED** [review]
Are there any tests we can run? `ctest`?

**Comment by hhslepicka** [comment]
@isuruf I can't run a test command as this software has no -help or similar so on the tests I am just checking for the existence of the binary. This code simply runs the input in a folder so that is basically all I can do other than running the input which may take several minutes.

**Pull request merged**